src/ecephys_etl/modules/vbn_create_stimulus_table/create_stim_table.py

```diff
@@ -0,0 +1,551 @@
+    median_stim_frame_duration = np.nanmedian(
+        stim_df["end_frame"] - stim_df["start_frame"]
+    )
+    omitted_end_frames = (
+        stim_df[stim_df['omitted']]['start_frame'] + median_stim_frame_duration
+    )
+    stim_df.loc[stim_df['omitted'], 'end_frame'] = omitted_end_frames
+
+    # Now fill in 'stop_time's for omitted stimuli
+    omitted_end_times = (
+        epoch_timestamps[stim_df[stim_df['omitted']]['end_frame'].astype(int)]
+    )
+    stim_df.loc[stim_df['omitted'], 'stop_time'] = omitted_end_times
+
+    # Drop index column
+    stim_df.drop(columns=["index"], inplace=True)
+
+    column_dtypes = {
+        "duration": float,
+        "end_frame": int,
+        "image_name": object,
+        "omitted": bool,
+        "orientation": float,
+        "start_frame": int,
+        "start_time": float,
+        "stop_time": float,
+        "stimulus_block": int,
+        "stimulus_name": object,
+        "is_change": bool,
+        "rewarded": bool,
+        "flashes_since_change": int,
+        "active": bool
+    }
+
+    return stim_df.astype(column_dtypes)
+
+
+def check_behavior_and_replay_pkl_match(
+    replay_pkl: ReplayPickleFile,
+    behavior_stim_table: pd.DataFrame
+):
+    ims = replay_pkl.image_presentations
+
+    # Check that replay pkl matches behavior
+    im_ons = []
+    im_offs = []
+    im_names = []
+    for ind, im in enumerate(ims):
+        if ind == 0:
+            continue
+        elif ind < len(ims) - 1:
+            if ims[ind - 1] is None and ims[ind] is not None:
+                im_ons.append(ind)
+                im_names.append(im)
+            elif ims[ind] is not None and ims[ind + 1] is None:
+                im_offs.append(ind)
+
+    inter_flash_interval = np.diff(im_ons)
+    putative_omitted = np.where(inter_flash_interval > 70)[0]
+
+    # Must specify dtype=object otherwise numpy will try to over-optimize
+    # by using a length limited character type (e.g. <U1, <U2, etc...)
+    # This causes problems when trying to insert "omitted" as it will may be
+    # accidentally truncated to "o", "om", etc...
+    im_names_arr = np.array(im_names, dtype=object)
+    im_names_with_omitted = np.insert(
+        im_names_arr, putative_omitted + 1, 'omitted'
+    )
+
+    # Handle omitted flash edge cases ###
+    behavior_df_image_names = behavior_stim_table['image_name']
+
+    # Check if the first flash was omitted
+    first_flash_omitted = behavior_df_image_names.iloc[0] == 'omitted'
+    if first_flash_omitted:
+        im_names_with_omitted = np.insert(im_names_with_omitted, 0, 'omitted')
+
+    # Check if last flash was omitted
+    last_flash_omitted = behavior_df_image_names.iloc[-1] == 'omitted'
+    if last_flash_omitted:
+        im_names_with_omitted = np.insert(
+            im_names_with_omitted,
+            len(im_names_with_omitted),
+            'omitted'
+        )
+
+    # Verify that the image list for replay is identical to behavior
```

Comment:
would be nice to add messages to these asserts to tell the user what happened

Reply:
Added